bundles/binding/org.openhab.binding.denon/src/main/java/org/openhab/binding/denon/internal/DenonBinding.java

```diff
@@ -137,73 +137,75 @@ protected void removeBindingProvider(DenonBindingProvider bindingProvider) {
      */
     @Override
     public void updated(Dictionary<String, ?> config) throws ConfigurationException {
-        if (config != null) {
-            logger.debug("Denon binding updated");
+        logger.debug("Denon binding updated");
 
-            Enumeration<String> keys = config.keys();
+        if (config == null) {
+            logger.warn("Denon binding has no configuration defined. Check the configuration file.");
```

Comment:
> However I don't understand why we're claiming it's not needed. If udpated() is passed a null config, the binding will not work. Which means it will fail silently.

Yes, but it will probably also be called with a non-`null` configuration around the time that the service is registered.

A ManagedService can have its `updated` method called by the ConfigAdminManager with a `null` dictionary, but its doing so doesn't mean there is really no configuration.  So it should be treated as a "nothing" and ignored.  I don't know if this is an implementation quirk or if there is some intended meaning behind it, but in practice, the only calls to `updated` that should do anything are ones where the `Dictionary` is not `null`.